mixins/form.jsx

```diff
@@ -411,7 +417,13 @@ module.exports = {
         submitting: true
       });
       submitProps = form.buildProps(props);
-      this.submit("/api/signup", submitProps, this.signupSuccess, this.signupError);
+      this.submit(url, submitProps, success, error);
     }
+  },
+  signup: function(validate, props) {
+    this.doSignup("/api/signup", validate, props, this.signupSuccess, this.signupError);
```

Comment:
I was actually thinking about a single route that can handle both services and you can maybe pass the data in such a way that won't break how we're currently doing it for `ET`

If you make a call to `/api/signup` from thunderbird which uses MailChimp
```
{
  service: 'mailchimp'
  data: {
    ...
    ...
  }
}
```
If you make a call to `/api/signup` from our regular donate page which uses ExactTarget
```
{
  service: 'basket'
  data: {
    ...
    ...
  }
}
```

That way in your route handler you can simply call the right method to handle the data from there, and for the `schema` since both form is very similar I think we can get away with having the server deal with that or just let the client pass everything we need there. Though `Joi` got a method that can deal with schemas differently here as well.